bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/ladda/LaddaAjaxButton.java

```diff
@@ -115,7 +104,9 @@ protected void updateAjaxAttributes(AjaxRequestAttributes attributes) {
     @Override
     protected <L extends Serializable> Component newLabel(String markupId, IModel<L> model) {
         Component label = super.newLabel(markupId, model);
-        label.add(AttributeModifier.append("class", "ladda-label"));
+        if (model.getObject() == null || "".equals(model.getObject())) {
```

Comment:
I would do Strings.isEmpty(model.getObject().toString())

Reply:
`Strings.isEmpty(model.getObject())` is not applicable to type `<L>` .....

Reply:
mind the getObject().toString

Reply:
This will be NPE in case of `null` :(((

Reply:
Will not model.getObject() == null protect you against that?

Reply:
https://stackoverflow.com/questions/17923872/condition-evaluation-process-in-java

Reply:
So you are proposing to replace 
`if (model.getObject() == null || "".equals(model.getObject()))`
with
`if (model.getObject() == null || Strings.isEmpty(model.getObject().toString())`

`Strings.isEmpty` will perform one more check for ` == null` then will compare with `""`

So new code will be longer
And will do 3 checks instead of 2

I can do it, but see no value in such change :(((

Reply:
The reason why I suggested this was another: what happens if getObject return "    "?

Reply:
@reiern70 the idea was to implement this https://getbootstrap.com/docs/4.4/components/spinners/#buttons "label free" button
Since checking the model is so questionable maybe it worth to add method like `withNoLabel` and use for checks?

Reply:
My only point was if you don't call trim in resulting string then using this as condition might surprise users as an string with one blank spaces will not satisfy your condition, even when nothing will be displayed as label.